### PR TITLE
Desktop Error handling and other hot fix

### PR DESCRIPTION
**Pull request by jigar-f**
- Added error handling for multiple methods
- Fixed issue with report Issue
- Use compute to avoid UI Freeze
- Fixed issue Device not showing up.

### CODE REVIEW

File: lib/common/session_model.dart [hunk 1]
```diff
@@ -247,11 +249,17 @@ class SessionModel extends Model {
   }
 
   Devices devicesFromJson(dynamic item) {
-    final items = item as List<dynamic>;
-    if (items.length == 0) return Devices.create();
-
-    final res = jsonEncode(items);
-    return Devices.create()..mergeFromProto3Json(jsonDecode(res));
+    final devices = <Device>[];
+    for (final element in item) {
```
Comment (jigar-f) [inline lib/common/session_model.dart:253]: Here I was getting an error that says  Json is expected, I ran a loop and passed json to protobuf

File: lib/common/session_model.dart [hunk 2]
```diff
@@ -546,9 +554,7 @@ class SessionModel extends Model {
   Future<void> reportIssue(
       String email, String issue, String description) async {
     if (isDesktop()) {
-      await ffiReportIssue(email.toNativeUtf8(), issue.toNativeUtf8(),
-          description.toNativeUtf8());
-      return;
+      return await compute(ffiReportIssue, [email, issue, description]);
```
Comment (jigar-f) [inline lib/common/session_model.dart:557]: I added compute, without this, the UI was freezing until the call completes. I think we need need to do other places as well.

**Comment by jigar-f** [comment]
Tagging @atavism , Making sure, You are seeing this...

File: desktop/lib.go [hunk 3]
```diff
@@ -248,14 +257,21 @@ func emailExists(email *C.char) *C.char {
...
+// If an error occurs during redemption, the first string is nil, and the second string contains the error message.
+//
 //export redeemResellerCode
-func redeemResellerCode(email, currency, deviceName, resellerCode *C.char) *C.char {
+func redeemResellerCode(email, currency, deviceName, resellerCode *C.char) (*C.char, *C.char) {
 	_, err := proClient.RedeemResellerCode(userConfig(), C.GoString(email), C.GoString(resellerCode),
 		C.GoString(deviceName), C.GoString(currency))
 	if err != nil {
-		return sendError(err)
+		log.Debugf("DEBUG: error while redeeming reseller code: %v", err)
```
Comment (atavism) [inline desktop/lib.go:269]: Maybe use `log.Error` here?

File: lib/common/session_model.dart (hunk 1 above)
Comment (atavism) [reply lib/common/session_model.dart:253]: I made my user Pro via the panel, but I also have no devices so I'm not sure I have any way of testing this. Did you confirm you see the device list when you navigate to the Pro account screen?

image

File: lib/ffi.dart [hunk 4]
```diff
@@ -86,14 +93,33 @@ Pointer<Utf8> ffiOnBoardingStatus() =>
...
+
+String ffiPaymentRedirect(planID, currency, provider, email, deviceName) {
+  final result =
+      _bindings.paymentRedirect(planID, currency, provider, email, deviceName);
+  if (result.r1 != nullptr) {
+    // Got error throw error to show error ui state
+    final errorCode = result.r1.cast<Utf8>().toDartString();
+    throw PlatformException(
+        code: errorCode,
+        message: 'we_are_experiencing_technical_difficulties'.i18n);
```
Comment (atavism) [inline lib/ffi.dart:119]: Nice, thanks for making this change

File: lib/account/report_issue.dart [hunk 5]
```diff
@@ -226,9 +227,8 @@ class _ReportIssueState extends State<ReportIssue> {
         },
       );
     } catch (error, stackTrace) {
-      if (Platform.isIOS) {
-        AppLoadingDialog.dismissLoadingDialog(context);
-      }
+      AppLoadingDialog.dismissLoadingDialog(context);
```
Comment (atavism) [inline lib/account/report_issue.dart:230]: Whenever I report an issue now, I am seeing a loading dialog with a Lantern logo:

Screenshot 2024-03-26 at 7 09 03 AM

Is that expected?

File: lib/common/session_model.dart (hunk 1 above)
Comment (jigar-f) [reply lib/common/session_model.dart:253]: Yes,I did, I see my devices on screen

File: lib/account/report_issue.dart (hunk 5 above)
Comment (jigar-f) [reply lib/account/report_issue.dart:230]: Yes, That is expected

**Pull request merged**